src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/Base/HardLinks/BaseHardLinks.FileSystem.cs

```diff
@@ -42,7 +43,11 @@ public void CreateHardLink_TargetDoesNotExist_Throws()
         {
             string linkPath = GetRandomFilePath();
             string nonExistentTarget = GetRandomFilePath();
-            Assert.Throws<FileNotFoundException>(() => CreateHardLink(linkPath, nonExistentTarget));
+            Assert.Throws<FileNotFoundException>(() =>
+            {
+                CreateHardLink(linkPath, nonExistentTarget);
+                try { File.Delete(linkPath); } catch { }
+            });
```

Comment:
```suggestion
            Assert.Throws<FileNotFoundException>(() => CreateHardLink(linkPath, nonExistentTarget));
```
you don't need to do this, the base class cleans up the directory on [dispose](https://github.com/dotnet/runtime/blob/5fc165a936c11ab22e1a242f1ea87abe70a683b2/src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs#L67).